pkg/network/encoding/json.go

```diff
@@ -17,12 +17,20 @@ type jsonSerializer struct {
 
 func (j jsonSerializer) Marshal(conns *network.Connections) ([]byte, error) {
 	agentConns := make([]*model.Connection, len(conns.Conns))
+	domainSet := make(map[string]int)
+
 	for i, conn := range conns.Conns {
-		agentConns[i] = FormatConnection(conn)
+		agentConns[i] = FormatConnection(conn, domainSet)
+	}
+
+	domains := make([]string, len(domainSet))
+	for k, v := range domainSet {
+		domains[v] = k
 	}
 
 	payload := connsPool.Get().(*model.Connections)
 	payload.Conns = agentConns
+	payload.Domains = domains
```

Comment:
Will we be encoding the same domains in FormatDNS and `payload.Domains` ?

Reply:
Yes, there will be some overlap but the sets won't always be the same. For example, domains that don't receive successful replies won't be be part of `payload.Dns` but they will be part of `payload.Domains`.